### PR TITLE
Fledgeling benchmark suite

### PR DESCRIPTION
**Pull request by jystic**
This just has the one stupid benchmark, but hopefully gets all the infrastructure in place for us to add more things over time.

### CODE REVIEW

File: src/Icicle/Benchmark.hs [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+  first BenchSourceError $ do
+    let inlined = sourceInline dict virtual
+
+    desugared    <- sourceDesugarQT inlined
+    (checked, _) <- sourceCheckQT dict desugared
+
+    let reified = sourceReifyQT checked
+
+    core <- sourceConvert dict reified
+    let simplified = coreSimp core
```
Comment (tranma) [inline src/Icicle/Benchmark.hs]: this (and above) is a bit like Pipeline stuff? Is it worth putting them in the main pipeline for repl and tests to use?

**Comment by tranma** [comment]
the generators are super cool @jystic, they could even be useful outside of icicle!
:100:

File: src/Icicle/Benchmark.hs (hunk 1 above)
Comment (jystic) [inline src/Icicle/Benchmark.hs]: yeah I think so, probably could simplify ice as well actually

Comment (tranma) [inline src/Icicle/Benchmark.hs]: yeh I saw some of these exact functions in ice

**Pull request merged**